### PR TITLE
*: support common aggregation functions elimination.

### PR DESCRIPTION
**Pull request by hanfei1991**
For query "select sum(a), sum(b) , sum(a) + sum(b) from t;", currently we put sum(a), sum(b), sum(a), sum(b) to aggregation. Instead, we should only consider sum(a) and sum(b).

@shenli @coocood @XuHuaiyu PTAL

### CODE REVIEW

File: expression/aggregation.go [hunk 1]
```diff
@@ -62,6 +63,59 @@ type AggregationFunction interface {
 	SetContext(ctx map[string](*ast.AggEvaluateContext))
 }
 
+// EqualExpression checks whether the two expressions are equal.
```
Comment (XuHuaiyu) [inline expression/aggregation.go]: the two --> two

File: expression/aggregation.go [hunk 2]
```diff
@@ -62,6 +63,59 @@ type AggregationFunction interface {
...
+		y, ok := b.(*Column)
+		if !ok {
+			return false
+		}
+		return x.FromID == y.FromID && x.Position == y.Position
+	}
+	return true
+}
+
+// EqualAgg checks whether the two aggregation functions are equal.
```
Comment (XuHuaiyu) [inline expression/aggregation.go]: ditto

File: expression/expression.go [hunk 3]
```diff
@@ -452,3 +452,41 @@ func SplitCNFItems(onExpr Expression) []Expression {
 func SplitDNFItems(onExpr Expression) []Expression {
 	return splitNormalFormItems(onExpr, ast.OrOr)
 }
+
+// EqualExpression checks whether the two expressions are equal.
+func EqualExpression(a, b Expression) bool {
```
Comment (XuHuaiyu) [inline expression/expression.go]: add a _Equals()_  function for Expression interface

File: expression/expression.go [hunk 4]
```diff
@@ -452,3 +452,41 @@ func SplitCNFItems(onExpr Expression) []Expression {
...
+		if err != nil || c != 0 {
+			return false
+		}
+	case *Column:
+		y, ok := b.(*Column)
+		if !ok {
+			return false
+		}
+		return x.FromID == y.FromID && x.Position == y.Position
+	}
```
Comment (tiancaiamao) [inline expression/expression.go]: default: return false ?

**Comment by coocood** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**